backup-environment/action.yml

```diff
@@ -5,7 +5,11 @@ description: 'Power Platform Backup Environment Action'
 inputs:
   environment-url:
     description: 'URL of Power Platform environment to backup; e.g. "https://test-env.crm.dynamics.com"'
-    required: true
+    required: false
+
+  environment-id:
+    description: 'ID of Power Platform environment to backup'
+    required: false
```

Comment:
It's not obvious at this top level that we require being provided either one or the other of these.  (Similar to it being non-obvious in PAC itself that we have an either/or-is-required constraint.)

We _could_ go the route of have two separate actions (eg, "backup-environment-by-id" and "backup-environment-by-url") where each has only the one option set as required.   I don't particularly like that option, but it would probably be the _clearest_ option.

Perhaps we should just add that as a note in the description of these two arguments?

Reply:
This is the same "mutually exclusive" properties pattern we already have with authN: username/password tuple vs tenantId/appId/secret triple. So splitting actions now by how we identify a target env (url vs envId) seems arbitrary and ultimately won't scale, as we'd end-up with too many permutations for each action. I posit separation of actions is a non-starter.
A short-term mitigation is what we already do for the authN tuple/triple:
- making it explicitly clear in the action.yml description, e.g. see:
https://github.com/microsoft/powerplatform-actions/blob/419e7d4da795b468aa584794ed0ed9153f34bc08/copy-environment/action.yml#L15
- and within an action, we assert that only one of the mutually exclusive sets is specified, and error out if overlapping info is specified (note: this is NOT a semantic check if the overlapping info agrees or conflicts, just the mere presence of multiple values is checked)

short term, we can follow that same pattern.
Moving forward, we have some options we can discuss:
a) merge env-url and env-id on the action level into a single string-typed 'environment' input parameter: within the action, we check/parse to see if the supplied string matches a guid format (-> env-id) or a url (-> env-url). We can then try to resolve this e.g. via PP discovery service if either is a match (heck, we could even just accept the string to match with friendly name or even the domain name?)
Ideally, this would be implemented on the pac CLI level, though, as it is too chatty
We should discuss this offline with PM